src/components/modal/examples/ModalConnectedExamples.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import { IReduxAction, ReduxConnect } from '../../../utils/ReduxUtils';
+import { openModal, IModalActionPayload } from '../ModalActions';
+import { ModalConnected } from '../ModalConnected';
+import { ModalBody } from '../ModalBody';
+import { ModalFooter } from '../ModalFooter';
+import { ModalBackdropConnected } from '../ModalBackdropConnected';
+
+export interface IModalExamplesProps {
+  openModal?: (id: string) => void;
+}
+
+const mapStateToProps = () => {
+  return {};
+};
+
+const mapDispatchToProps = (dispatch: (action: IReduxAction<IModalActionPayload>) => void): IModalExamplesProps => {
+  return {
+    openModal: (id: string) => {
+      dispatch(openModal(id));
+    }
+  };
+};
+
+@ReduxConnect(mapStateToProps, mapDispatchToProps)
+export class ModalConnectedExamples extends React.Component<IModalExamplesProps, any> {
+
+  openModal(id: string) {
+    if (this.props.openModal) {
```

Comment:
Since `ModalConnectedExamples` is always connected, you can skip this and use `this.props.openModal('modal2')` in your onClick functions.